azure-iot-device/azure/iot/device/iothub/pipeline/pipeline_ops_iothub.py

```diff
@@ -6,6 +6,30 @@
 from azure.iot.device.common.pipeline import PipelineOperation
 
 
+class SetX509AuthProviderOperation(PipelineOperation):
+    """
+    A PipelineOperation object which tells the pipeline to use a particular X509 authorization provider.
+    Some pipeline stage is expected to extract arguments out of the auth provider and pass them
+    on so an even lower stage can use those arguments to connect.
+
+    This operation is in the group of IoTHub operations because authorization providers are currently
+    very IoTHub-specific
+    """
+
+    def __init__(self, auth_provider, callback=None):
+        """
+        Initializer for SetAuthProviderOperation objects.
+
+        :param object auth_provider: The X509 authorization provider object to use to retrieve connection parameters
+          which can be used to connect to the service.
+        :param Function callback: The function that gets called when this operation is complete or has failed.
+         The callback function must accept A PipelineOperation object which indicates the specific operation which
+         has completed or failed.
+        """
+        super(SetX509AuthProviderOperation, self).__init__(callback=callback)
+        self.auth_provider = auth_provider
+
+
 class SetAuthProviderOperation(PipelineOperation):
```

Comment:
Add a comment (TODO) that indicates SetAuthProviderOperation is really an operation that uses a SAS token for auth, and that SetX509AuthProviderOperation is really an operation that uses certs for auth